var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -407,14 +407,16 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
     # ###################### Dependencies ##########################
 
     # External Kokkos
-    depends_on("kokkos@4.4.01~complex_align", when="@master: +kokkos")
-    depends_on("kokkos@4.3.01~complex_align", when="@16.0:16 +kokkos")
-    depends_on("kokkos@4.2.01~complex_align", when="@15.1:15 +kokkos")
-    depends_on("kokkos@4.1.00~complex_align", when="@14.4:15.0 +kokkos")
-    depends_on("kokkos-kernels@4.4.01", when="@master: +kokkos")
-    depends_on("kokkos-kernels@4.3.01", when="@16.0:16 +kokkos")
-    depends_on("kokkos-kernels@4.2.01", when="@15.1:15 +kokkos")
-    depends_on("kokkos-kernels@4.1.00", when="@14.4:15.0 +kokkos")
+    with when("@14.4: +kokkos"):
+        depends_on("kokkos~complex_align")
+        depends_on("kokkos@4.4.01", when="@master:")
```

Comment:
should this then be `@17:` instead of `@master:` to cover the full range of possible versions?